src/hotspot/cpu/x86/stubGenerator_x86_64_arraycopy.cpp

```diff
@@ -1108,7 +1220,7 @@ __ BIND(L_exit);
   {
     UnsafeCopyMemoryMark ucmm(this, !aligned, false, ucme_exit_pc);
     // Copy in multi-bytes chunks
-    copy_bytes_forward(end_from, end_to, qword_count, rax, L_copy_bytes, L_copy_8_bytes);
+    copy_bytes_forward(end_from, end_to, qword_count, rax, r10, L_copy_bytes, L_copy_8_bytes, decorators, T_BYTE);
```

Comment:
Why is `setup_arg_regs_using_thread()` not needed to free `r10` as a temporary register in `generate_[conjoint|disjoint]_[byte|short]_copy()`? From the following comment, it sounds like it would be needed:

https://github.com/openjdk/jdk/blob/7d8b8ba9c46475ededcae7db6c841b25fa83d167/src/hotspot/cpu/x86/stubGenerator_x86_64.cpp#L1193-L1195

Reply:
Thanks for the explanation, Erik! If I got it right, in this patch, implementations of `copy_[load|store]_at()` can only use `r10` safely (`Register tmp`, or `Register tmp2` for vector `copy_store_at()`) if `BasicType type == T_OBJECT`, because only in this context they will be called between `setup_arg_regs_using_thread()` and `restore_arg_regs_using_thread()` (as opposed to regular `setup_arg_regs()`/`restore_arg_regs()`). If this is correct and expected, I suggest to document it somewhere, e.g. as a comment in the declaration of `copy_[load|store]_at()`, to avoid misinterpretations.

Reply:
Good point. I'll add a comment.